### PR TITLE
Revert replication tests

### PR DESCRIPTION
**Pull request by antoniofilipovic**
Replication tests and multi-tenancy tests started failling because now, on replica restart we need to recover replication role together with main's UUID. This PR intends to fix that. Also, there was one bug on CreateDatabaseRPC, main_uuid was not serialized.

[master < Task] PR
- [ ] Check, and update documentation if necessary
- [ ] Provide the full content or a guide for the final git message

To keep docs changelog up to date, one more thing to do:
- [ ] Write a release note here, including added/changed clauses
- [ ] Tag someone from docs team in the comments

### CODE REVIEW

File: tests/e2e/replication_experimental/multitenancy.py [hunk 1]
```diff
@@ -438,7 +456,29 @@ def test_automatic_databases_create_multitenancy_replication(connection):
...
+    )
+
+    main_cursor = connection(BOLT_PORTS["main"], "main").cursor()
+
+    execute_and_fetch_all(
+        main_cursor, f"REGISTER REPLICA replica_1 SYNC TO '127.0.0.1:{REPLICATION_PORTS['replica_1']}';"
+    )
+    execute_and_fetch_all(
+        main_cursor, f"REGISTER REPLICA replica_2 ASYNC TO '127.0.0.1:{REPLICATION_PORTS['replica_2']}';"
+    )
```
Comment (antoniofilipovic) [inline tests/e2e/replication_experimental/multitenancy.py:480]: I can isolate this into one function, TODO

File: src/replication_handler/replication_handler.cpp [hunk 2]
```diff
@@ -198,6 +198,7 @@ bool ReplicationHandler::SetReplicationRoleReplica(const memgraph::replication::
                                                    const std::optional<utils::UUID> &main_uuid) {
   // We don't want to restart the server if we're already a REPLICA
```
Comment (as51340) [inline src/replication_handler/replication_handler.cpp:199]: if you log, comment probably not needed

**Review by as51340 — APPROVED** [review]

File: tests/e2e/replication_experimental/multitenancy.py [hunk 3]
```diff
@@ -985,7 +1155,29 @@ def test_multitenancy_drop_and_recreate_while_replica_using(connection):
...
+        replica_2_cursor, f"SET REPLICATION ROLE TO REPLICA WITH PORT {REPLICATION_PORTS['replica_2']};"
+    )
+
+    main_cursor = connection(BOLT_PORTS["main"], "main").cursor()
+
+    execute_and_fetch_all(
+        main_cursor, f"REGISTER REPLICA replica_1 SYNC TO '127.0.0.1:{REPLICATION_PORTS['replica_1']}';"
+    )
+    execute_and_fetch_all(
+        main_cursor, f"REGISTER REPLICA replica_2 ASYNC TO '127.0.0.1:{REPLICATION_PORTS['replica_2']}';"
```
Comment (andrejtonev) [inline tests/e2e/replication_experimental/multitenancy.py:1178]: Because of the flag, we keep doing this manually. Can you put them in a function?

**Review by andrejtonev — APPROVED** [review]

File: src/replication_handler/replication_handler.cpp (hunk 2 above)
Comment (antoniofilipovic) [reply src/replication_handler/replication_handler.cpp:199]: This will be handled in this PR: https://github.com/memgraph/memgraph/pull/1711

**Comment by andrejtonev** [comment]
@antoniofilipovic I think we need the docs here. 
@kgolubic one of the PRs before made it necessary to use the `--replication-restore-state-on-startup` flag. This PRs just makes the tests pass, but it is important to document the new behavior.

**Comment by kgolubic** [comment]
@antoniofilipovic I've added the tag that docs are needed.

**Comment by antoniofilipovic** [comment]
@andrejtonev did you add docs here? I saw on documentation PR that this part is described. 

So docs are added here: https://github.com/memgraph/documentation/pull/476

**Comment by antoniofilipovic** [comment]
@kgolubic RN: When using replication, flag `--replication-restore-state-on-startup` needs to be set to `true`.

**Pull request merged**